### PR TITLE
feature: expose GalleryItem

### PR DESCRIPTION
**Pull request by marlonmarcello**
# Goal

Right now there is no way to target the `<li>` item that is auto generated and wrapped around the render function. So trying to customize it while using CSS modules require the use of the `:global` mixin.

The goal of this PR is to expose the `<GalleryItem />` component to allow for further customization of the automatically rendered `<li>` tag.

This is a breaking change so I bumped the version to `2.0.0`.

## Usage

Not much changes, only things is that now users need to return a `<GalleryItem />` component inside their render function:

```jsx
<Gallery items={GALLERY_ITEMS}>
    <GalleryMain
      renderGalleryItem={({ item, i, active }) => (
        <GalleryItem key={i} index={i} active={active}>
          <img src={item.image} alt={item.alt} />
        </GalleryItem>
      )}
    />
</Gallery>
```

### CODE REVIEW

**Review by liamegan — DISMISSED** [review]
LGTM

**Comment by andrewrubin** [comment]
This works for me! Thanks for the update. Do you mind updating the docs as well? No rush.

If this is not urgent, I think we should prep a release before merging this to main, which also includes a fix to #3.

File: README.md [hunk 1]
```diff
@@ -106,7 +108,7 @@ export default YourGallery
 
 The first step is to give your data to the `<Gallery>` component via the `items` prop. At the very least, `items` is expected to be an Array. From there, you're free to arrange the child components this package provides as you see fit. Below is a brief description of each of the child components' usage. For a detailed breakdown of this component, jump ahead to the [Gallery](#gallery) section.
 
-`<GalleryMain>` is the primary gallery view where your item data is rendered. It receives a render prop, `renderGalleryItem`, which exposes a few arguments you can use in the JSX you return: `item`, `i`, `activeIndex`, and `active`. For a detailed breakdown of this component, jump ahead to the [GalleryMain](#gallerymain) section.
+`<GalleryMain>` is the primary gallery view where your item data is rendered. It receives a render prop, `renderGalleryItem`, which exposes a few arguments you can use in the JSX you return: `item`, `i`, `activeIndex`, and `active` and expectes a `<GalleryItem>` to be returned. For a detailed breakdown of this component, jump ahead to the [GalleryMain](#gallerymain) section.
```
Comment (andrewrubin) [inline README.md:111]: Small typo on `expects`

File: README.md [hunk 2]
```diff
@@ -76,15 +76,17 @@ export const GALLERY_ITEMS = [
...
 
 const YourGallery = () => {
   return (
     <Gallery items={GALLERY_ITEMS}>
 
       <GalleryMain
-        renderGalleryItem={({ item }) => (
-          <img src={item.image} alt={item.alt} />
+        renderGalleryItem={({ item, i, active }) => (
+          <GalleryItem key={i} i={i} active={active}>
```
Comment (andrewrubin) [inline README.md:87]: I think the prop here is called `index`, not `i`, right?

**Review by andrewrubin — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -187,6 +189,17 @@ This render prop wraps its return value in a list item (`<li>`), and receives a
 | i           | Number  | The index of the current item being iterated over.                                                             |
 | item        | Any     | The current item being iterated over, as defined by the Array fed to the `<Gallery>` component's `items` prop. |
 
+### &lt;GalleryItem&gt;
```
Comment (andrewrubin) [inline README.md:192]: Can we also add this to the table of contents at the top of the README?

File: README.md (hunk 2 above)
Comment (marlonmarcello) [reply README.md:87]: Correct, fixed!

File: README.md (hunk 1 above)
Comment (marlonmarcello) [reply README.md:111]: Thank you!

File: README.md (hunk 3 above)
Comment (marlonmarcello) [reply README.md:192]: Done!

**Review by andrewrubin — APPROVED** [review]

**Pull request merged**